src/ATDPApplication.ts

```diff
@@ -126,6 +137,28 @@ export abstract class ATDPApplication<T> extends ACLUEWrapper {
     });
   }
 
+  /**
+   * Loads the client config from '/clientConfig.json' and parses it.
+   */
+  public static loadClientConfig<T = any>(): Promise<T | null> {
+    return Ajax.getJSON('/clientConfig.json').catch((e) => {
+      // TODO: Do you want to print an error here, or should it fail silently?
+      console.error('Error parsing clientConfig.json', e);
+      return null;
+    });
+  }
+
+  /**
+   * Loads the client config via `loadClientConfig` and automatically merges it into the options.
+   * @param options Options where the client config should be merged into.
+   */
+  public static initializeClientConfig(options: ITDPOptions): Promise<ITDPOptions> {
+    return ATDPApplication.loadClientConfig().then((parsedConfig) => {
+      options.clientConfig = BaseUtils.mixin(options?.clientConfig || {}, parsedConfig || {});
```

Comment:
```suggestion
      options.clientConfig = BaseUtils.mixin(options?.clientConfig, parsedConfig || {});
```

Why not initializing the client config already above? Then we could save this error fallback.

Reply:
It is still necessary, because the function might be called from outside then the `options.clientConfig` is `null`.